outlines/models/llamacpp.py

```diff
@@ -166,7 +166,9 @@ def prepare_generation_parameters(
 
         # Somehow `llama-cpp-python` generates `max_tokens + 1`  tokens
         if "max_tokens" not in llama_cpp_params:
-            if max_tokens is not None:
+            if max_tokens is None:
```

Comment:
Did you check `llama.cpp`’s code? They could be using `-1` to indicate there’s no limit.

Reply:
Great catch, `llama.cpp`'s behavior.